### PR TITLE
[Static reference] LPS-182671 Remove static references in class MBDiscussionPermission | part 1

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by catarineaguiar** [comment]
ci:test:lima

**Comment by liuxingye** [comment]
ci:test:stable

File: modules/apps/comment/comment-test/src/testIntegration/java/com/liferay/comment/subscriptions/test/CommentsSubscriptionTest.java [hunk 1]
```diff
@@ -101,6 +101,8 @@ public void testCannotSubscribeToCommentsWithoutViewPermissions()
 
 		Assert.assertFalse(
 			discussionPermission.hasSubscribePermission(
+				PermissionCheckerFactoryUtil.create(
+					UserLocalServiceUtil.getGuestUser(_group.getCompanyId())),
 				TestPropsValues.getCompanyId(), _group.getGroupId(),
```
Comment (yuhai) [inline modules/apps/comment/comment-test/src/testIntegration/java/com/liferay/comment/subscriptions/test/CommentsSubscriptionTest.java:100]: @W2jx, in this commit (https://github.com/liferay-core-infra/liferay-portal/blob/593c08b4f414dc5bbcddb96211aec620ad9d1430/modules/apps/comment/comment-test/src/testIntegration/java/com/liferay/comment/subscriptions/test/CommentsSubscriptionTest.java#L97-L107)

`PermissionCheckerFactoryUtil.create(UserLocalServiceUtil.getGuestUser(_group.getCompanyId())` was used twice in the same method,  Please extract one param for it.

File: modules/apps/message-boards/message-boards-comment/src/main/java/com/liferay/message/boards/comment/internal/MBDiscussionPermissionImpl.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import com.liferay.portal.kernel.exception.PortalException;
 import com.liferay.portal.kernel.security.permission.ActionKeys;
 import com.liferay.portal.kernel.security.permission.PermissionChecker;
+
 import org.osgi.service.component.annotations.Component;
```
Comment (yuhai) [inline modules/apps/message-boards/message-boards-comment/src/main/java/com/liferay/message/boards/comment/internal/MBDiscussionPermissionImpl.java:27]: @W2jx, this is SF commit, please merge it to the  appropriate commit.

**Comment by yuhai** [comment]
@W2jx, I have two suggestions.

1. Please merge the commit https://github.com/liferay-core-infra/liferay-portal/pull/4944/commits/7a811c9a05e180c5115721caa48b37762eb5e0a4 and https://github.com/liferay-core-infra/liferay-portal/pull/4944/commits/209562c792ddf499577085e66a9d5f4ec56ec401
The commit message may be "Create new DiscussionPermissionUtil class to get DiscussionPermission, instead of using CommentManagerUtil, prepare for next"

2. Edit the [commit ](https://github.com/liferay-core-infra/liferay-portal/pull/4944/commits/96d7c815224999c84e6f1a13da058612c9cd935b)message for "remove unnecessary constructor". And then separate the [commit](https://github.com/liferay-core-infra/liferay-portal/pull/4944/commits/f8a40364b5a2d6a19f799ff24a7ed5d38a5b5a8d) for two commits.
One is "update usages" in MBCommentManagerImpl, CommentManagerUtil and CommentManager.
Another is "update CommentManager related usages"

**Comment by W2jx** [comment]
I will resend it.

**Pull request closed without merging**